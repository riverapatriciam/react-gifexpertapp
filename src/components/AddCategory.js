import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({categories = [], setCategories}) => {

  const [inputValue, setinputValue] = useState('');

  const handleInputChange = (e) => {
    setinputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {

        //Forma mas eficiente, ya que no hay que mandar los categories como props
        //setCategories ( cats => [...categories, inputValue])

        setCategories([inputValue, ...categories]);
        setinputValue('');
    }

  };

  return (
    <>
      <form onSubmit={ handleSubmit }>
        <p>{inputValue}</p>
        <input 
            type="text" 
            value={inputValue} 
            onChange={handleInputChange} />
      </form>
    </>
  );
};

AddCategory.protoTypes = {
    setCategories: PropTypes.func.isRequired
}
 