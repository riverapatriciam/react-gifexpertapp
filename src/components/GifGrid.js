import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';


export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs(category);
    
  
    return (
    <>
        <h3>{category}</h3>

        {loading && 'Cargando...' }

        <div className='card-grid'>
            <ol>
                {
                    data.map( img => ( 
                        <GifGridItem 
                            key={img.id}
                            {...img} 
                        />
                    ))
                }
            </ol>
        </div>
    </>
    
  )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}