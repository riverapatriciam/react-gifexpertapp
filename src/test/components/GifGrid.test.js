import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componete <GifGrid />', () => { 

    const category = 'FMA';
    
    test('Debe mostrar correctamente el componente', () => { 
        
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();

     })

     test('Debe mostrar items cuando se cargan imagenes de useFecthGifs', () => { 

        const gifs = [{
            id: 'ABC',
            title: 'Title',
            url: 'http://localhost/title.jpj'
        }];
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        


      })

 })