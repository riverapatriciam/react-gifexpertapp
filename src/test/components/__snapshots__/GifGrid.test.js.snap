// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componete <GifGrid /> Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h3>
    FMA
  </h3>
  Cargando...
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas en el componete <GifGrid /> Debe mostrar items cuando se cargan imagenes de useFecthGifs 1`] = `
<Fragment>
  <h3>
    FMA
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Title"
        url="http://localhost/title.jpj"
      />
    </ol>
  </div>
</Fragment>
`;
